src/dune_engine/build_system.ml

```diff
@@ -2145,13 +2159,15 @@ end = struct
      as its digest. *)
   let build_file_impl path =
     let t = t () in
-    let on_error exn = Dep_path.reraise exn (Path path) in
-    Memo.Build.with_error_handler ~on_error (fun () ->
+    Memo.push_stack_frame
+      (fun () ->
         get_rule_or_source t path >>= function
         | Source digest -> Memo.Build.return digest
         | Rule (path, rule) ->
           let+ { deps = _; targets } = execute_rule rule in
           Path.Build.Map.find_exn targets path)
+      ~info:(fun () ->
+        { desc = Some (Pp.text (Dpath.describe_path path)); loc = None })
```

Comment:
In reference to previous comment: I'd like this information to be available in Build_system.Error.t. Preferably, as a raw path that I can serialize rather than a pretty printed string.

Reply:
That's fine, but first let's make sure that's actually what we want. If you want a source file, passing this along is not going to help.